.github/Dockerfile

```diff
@@ -0,0 +1,6 @@
+# Minimal Dockerfile that just wraps pre-built binaries, so we can test the server inside docker
+FROM rust:1.90.0
```

Comment:
there might be an even smaller container we can use as the base, but this one seems fine. Currently it takes ~39s to set up the docker container, and most of that is installing this base image.